### PR TITLE
Fix Date picker content when pane is splitted (#46900). Show narrow(m…

### LINKED ISSUE #46900: Date picker is partially hidden in Explore split pane view
**What happened**:

In Grafana Explore, with the split-pane view, when I click on the calendar button to select an absolute time range, the popup date picker is hidden behind the split-pane separator. When I try the same on the left pane, the date picker is partially hidden to the left of the browser viewport.

Date picker on the right pane:
Screenshot 2022-03-23 13-50-27 split pane bug

Date picker on the left pane:
Screenshot 2022-03-23 13-54-04 split pane bug

**What you expected to happen**:

The date picker should be positioned in a way that it is entirely visible on the screen.

**How to reproduce it (as minimally and precisely as possible)**:

1. Go to Grafana Explore.
2. Click the Split button.
3. Select an absolute time range on either pane using the date picker.

**Anything else we need to know?**:

**Environment**:
- Grafana version: v8.4.4
- Data source type & version:
- OS Grafana is installed on: Ubuntu 20.04
- User OS & Browser: Ubuntu 20.04, Firefox 98.0.1
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by glintik**
**Which issue(s) this PR fixes**:

Fixes #46900.

**Special notes for your reviewer**:

Actually, this is a workaround and the fix just takes into account pane split.
Really good and universal fix is to rework this UI element to make it unattached to pane. Instead, add dom elements to root divs. Also date picker can be much more mobile-friendly.

### CODE REVIEW

**Comment by glintik** [comment]
Hi @gabor,

Pardon, that PR was not from main branch forked. This one should be good.

**Comment by gelicia** [comment]
@glintik We discussed this internally and the UX team has reservations about the large amount of blank space that shows in the time picker drop down and the pop up for the calendar. Would you have the desire to look into this more? Otherwise this is something a Grafana team can look into - I'm planning on taking a look in the next few days to see if there's a way to stay with the low code impact you have (which is great!) and not have some of the UX concerns that were noted.

**Comment by glintik** [comment]
> We discussed this internally and the UX team has reservations about the large amount of blank space that shows in the time picker drop down and the pop up for the calendar. Would you have the desire to look into this more? Otherwise this is something a Grafana team can look into - I'm planning on taking a look in the next few days to see if there's a way to stay with the low code impact you have (which is great!) and not have some of the UX concerns that were noted.

Hi @gelicia,
May you please point out what UX team suggest? Mobile-style picker is not acceptable for splitted mode?

**Comment by gelicia** [comment]
> Hi @gelicia, May you please point out what UX team suggest? Mobile-style picker is not acceptable for splitted mode?

Their issue was with the big blank space on the side and the calendar modal that pops up.
Screen Shot 2022-04-12 at 5 28 57 PM

I had some time and made the changes I think would be good. I changed the `splitted` boolean to a `widthOverride` number because split is a concept in explore only and `widthOverride` is a little more universal. The issue with the big empty space on the time picker was due to a couple media queries in the component that were not seeing the `isFullscreen` boolean and adjusting accordingly. Finally, the calendar modal was showing up in an odd area and I changed it so it will always show up in the middle of the window - this may not be the best approach but at least it will be consistent.

File: packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker/TimePickerCalendar.tsx [hunk 1]
```diff
@@ -37,7 +37,6 @@ export const getStyles = (theme: GrafanaTheme2, isReversed = false) => {
     modal: css`
       position: fixed;
       top: 20%;
-      left: 25%;
```
Comment (gelicia) [inline packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker/TimePickerCalendar.tsx:40]: Ideally we would want this modal to pop up next to the button that called it, but this at least will have it show consistently in the middle of the window. I feel like the left position was put here for a reason, however, so let me know what my options might be short of a heavy refactor of the calendar modal.

File: packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker/TimePickerContent.tsx [hunk 2]
```diff
@@ -251,22 +251,18 @@ const useTimeOption = (raw: RawTimeRange, quickOptions: TimeOption[]): TimeOptio
...
-const getStyles = stylesFactory((theme: GrafanaTheme2, isReversed, hideQuickRanges, isContainerTall) => {
+const getStyles = stylesFactory((theme: GrafanaTheme2, isReversed, hideQuickRanges, isContainerTall, isFullscreen) => {
   return {
     container: css`
       background: ${theme.colors.background.primary};
       box-shadow: ${theme.shadows.z3};
       position: absolute;
       z-index: ${theme.zIndex.dropdown};
-      width: 546px;
+      width: ${isFullscreen ? '546px' : '262px'};
```
Comment (gelicia) [inline packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker/TimePickerContent.tsx:261]: I realize this is leaving out the `only screen` aspect of this, but I think this kind of styling would be universal to all media types.

**Comment by glintik** [comment]
@gelicia 
Thank you for your additions and adjustments. I spotted one issue - now timepicker is always mobile-size :). Please fix `widthOverride` value considering `splitted` mode.
As you noted, this timepicker should be reworked, it became too complex and too much interconnects by values/logic and mobile view is really not optimal.
So I'd recommend to put this component to "To be refactored/reworked" backlog or something similar, if possible. BTW, that's a good "First issue" also, Senior level.

File: public/app/features/explore/ExploreTimeControls.tsx [hunk 3]
```diff
@@ -92,6 +92,7 @@ export class ExploreTimeControls extends Component<Props> {
         {...timePickerCommonProps}
         timeSyncButton={timeSyncButton}
         isSynced={syncedTimes}
+        widthOverride={splitted ? window.innerWidth / 2 : undefined}
```
Comment (gelicia) [inline public/app/features/explore/ExploreTimeControls.tsx:95]: @glintik is this what you had in mind? this was my mistake - thanks for catching it!

**Comment by glintik** [comment]
Yep, now is perfect!

**Comment by ifrost** [comment]
Thanks! It works nicely, definitely a good improvement, so we can merge it 👍 It would be nice to have a look if we could make it more responsive, by somehow changing the `overflow` in the scrollbar container and making use of [`isReversed` from TimePickerContent](https://github.com/grafana/grafana/blob/v8.5.0/packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker/TimePickerContent.tsx#L30) to get something like this:

Screen Shot 2022-04-21 at 18 46 10

**Review by ifrost — APPROVED** [review]

**Review by codeincarnate — APPROVED** [review]

**Pull request merged**